File: 950-XOfAKindInADeckOfCards/950-XOfAKindInADeckOfCards.py
# Last updated: 8/13/2025, 8:18:18 PM
class Solution:
    def hasGroupsSizeX(self, deck: List[int]) -> bool:
        feq_map = {}

        for i in range(len(deck)) : 
            feq_map[deck[i]] = feq_map.get(deck[i] , 0) + 1 
        
        ans = 0
        for key in feq_map.keys() : 
            ans = self.gcdCalculator(ans , feq_map[key])
            if ans <=1 : 
                return False
        
        return True
    
    def gcdCalculator(self , a:int , b:int) -> int : 
        if a == 0 : 
            return b
        
        temp = self.gcdCalculator(b%a , a)
        return temp

File: 950-XOfAKindInADeckOfCards/950-XOfAKindInADeckOfCards.java
// Last updated: 8/13/2025, 8:18:24 PM
class Solution {
    public boolean hasGroupsSizeX(int[] deck) {
        int n = deck.length;
        HashMap<Integer,Integer> h = new HashMap();
        for(int i=0;i<n;i++){
            if(h.containsKey(deck[i])==false){
                h.put(deck[i],1);
            }
            else{
                int temp = h.get(deck[i]);
                h.put(deck[i],temp+1);
            }
        }
        int ans = 0;
        for(int key : h.keySet()){
            ans = gcd(h.get(key),ans);
        }
        if(ans==1){
            return false;
        }
        return true;
    }

    public int gcd(int a , int b){
        if(a==0){
            return b;
        }
        int temp = gcd(b%a,a);
        return temp;
    }
}